flink-core/src/main/java/org/apache/flink/util/FileUtils.java

```diff
@@ -107,6 +120,89 @@ public static void writeFileUtf8(File file, String contents) throws IOException
 		writeFile(file, contents, "UTF-8");
 	}
 
+	/**
+	 * Reads all the bytes from a file. The method ensures that the file is
+	 * closed when all bytes have been read or an I/O error, or other runtime
+	 * exception, is thrown.
+	 *
+	 * <p>This is an implementation that follow {@link java.nio.file.Files#readAllBytes(java.nio.file.Path)},
+	 * and the difference is that it limits the size of the direct buffer to avoid
+	 * direct-buffer OutOfMemoryError. When {@link java.nio.file.Files#readAllBytes(java.nio.file.Path)}
+	 * or other interfaces in java API can do this in the future, we should remove it.
+	 *
+	 * @param path
+	 *        the path to the file
+	 * @return a byte array containing the bytes read from the file
+	 *
+	 * @throws IOException
+	 *         if an I/O error occurs reading from the stream
+	 * @throws OutOfMemoryError
+	 *         if an array of the required size cannot be allocated, for
+	 *         example the file is larger that {@code 2GB}
+	 */
+	public static byte[] readAllBytes(java.nio.file.Path path) throws IOException {
+		try (SeekableByteChannel channel = Files.newByteChannel(path);
```

Comment:
Did you copy this code from [stackoverflow](https://stackoverflow.com/questions/43782158/java-read-a-big-file-few-gb-without-exception)?

Reply:
if so then we need property attribution for said content, according to their license:

Section 6 - Subscriber content:
https://stackoverflow.com/legal/terms-of-service/public

https://creativecommons.org/licenses/by-sa/4.0/

Reply:
Ah, thanks for the reminder @zentol !
This code is come from `java.nio.file.Files#readAllBytes`, the change only add the limitation of `DirectBufferSize` in the `read` method.  Do we also need to follow  [stackoverflow](https://stackoverflow.com/questions/43782158/java-read-a-big-file-few-gb-without-exception) license ?

Reply:
@sunjincheng121  is right.

This OutOfMemoryError is different from the [stackoverflow ](https://stackoverflow.com/questions/43782158/java-read-a-big-file-few-gb-without-exception) one. That one is on heap memory, while this one is on direct memory.

The codes are not copied from stackoverflow, they come from java.nio.file.Files#readAllBytes() and a little changes have been made. Does this have licensing issues? @zentol

Reply:
ah ok that makes sense, I didn't look into it deep enough.

Reply:
@zentol anyway thanks for your reminder, are there any other suggestions for you?  if not, I will merge this PR. :-)